warehouse/integrations/azure-synapse/azure-synapse.go

```diff
@@ -218,232 +222,394 @@ func (*AzureSynapse) IsEmpty(context.Context, model.Warehouse) (empty bool, err
 	return
 }
 
-func (as *AzureSynapse) loadTable(ctx context.Context, tableName string, tableSchemaInUpload model.TableSchema, skipTempTableDelete bool) (stagingTableName string, err error) {
-	as.logger.Infof("AZ: Starting load for table:%s", tableName)
-
-	previousColumnKeys := warehouseutils.SortColumnKeysFromColumnMap(as.Uploader.GetTableSchemaInWarehouse(tableName))
-	// sort column names
-	sortedColumnKeys := warehouseutils.SortColumnKeysFromColumnMap(tableSchemaInUpload)
+func (as *AzureSynapse) loadTable(
+	ctx context.Context,
+	tableName string,
+	tableSchemaInUpload model.TableSchema,
+	skipTempTableDelete bool,
+) (*types.LoadTableStats, string, error) {
+	log := as.logger.With(
+		logfield.SourceID, as.Warehouse.Source.ID,
+		logfield.SourceType, as.Warehouse.Source.SourceDefinition.Name,
+		logfield.DestinationID, as.Warehouse.Destination.ID,
+		logfield.DestinationType, as.Warehouse.Destination.DestinationDefinition.Name,
+		logfield.WorkspaceID, as.Warehouse.WorkspaceID,
+		logfield.Namespace, as.Namespace,
+		logfield.TableName, tableName,
+	)
+	log.Infow("started loading")
 
-	var extraColumns []string
-	for _, column := range previousColumnKeys {
-		if !slices.Contains(sortedColumnKeys, column) {
-			extraColumns = append(extraColumns, column)
-		}
-	}
 	fileNames, err := as.LoadFileDownLoader.Download(ctx, tableName)
-	defer misc.RemoveFilePaths(fileNames...)
+	defer func() {
+		misc.RemoveFilePaths(fileNames...)
+	}()
```

Comment:
Thanks for bringing this up.

Reply:
> Then looking at the internals it seems like Download is not cleaning up after itself. If there has been an error, all files that have been downloaded for nothing (and are not returned apparently) could be cleaned up by the downloader itself.

That is true, in case of an error it does not delete the files.

Reply:
I will move the defer block after the error. No point in keeping up before error check.

Reply:
Have a separate task and PR for this?

Reply:
Tracking it [here](https://linear.app/rudderstack/issue/PIPE-286/integrations-cleanup-download-task).